### PR TITLE
fix regex expression in SSH config parser

### PR DESCRIPTION
**Pull request by jeloou**
I've the following config in my `~/.ssh/config`:

```
Host sshd.docker0
  ForwardAgent yes
  StrictHostKeyChecking no
  Port 32770
  User joseph

Host sshd.docker1
  Port 32771
  User joseph
```

but this isn't parsed correctly by `ParseSSHConfigFile` because the first host has `StrictHostKeyChecking` config. 

this fix handles that scenario correctly.

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func ParseSSHConfigFile(path string) (map[string]SSHClientOptions, error) {
 
 	// Read lines in reverse order and parse option for each Host section:
 	lines := strings.Split(string(content), "\n")
-	hostExpr := regexp.MustCompile("\\s*Host\\s*=?\\s*(.+)")
+	hostExpr := regexp.MustCompile("\\s*Host\\s*(=?)\\s*(.+)")
```
Comment (mrkschan) [inline config.go]: I wonder if it's possible to just update this regexp as `"^\\s*Host\\b\\s*=?\\s*(.+)"` and we're done.

See https://play.golang.org/p/J8GdA1UOTv

Comment (jeloou) [reply config.go]: You right, that worked great. I pushed the changes. Thanks!

File: config.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func ParseSSHConfigFile(path string) (map[string]SSHClientOptions, error) {
 
 	// Read lines in reverse order and parse option for each Host section:
 	lines := strings.Split(string(content), "\n")
-	hostExpr := regexp.MustCompile("\\s*Host\\s*=?\\s*(.+)")
+	hostExpr := regexp.MustCompile("\\s*Host\\b\\s*=?\\s*(.+)")
```
Comment (mrkschan) [inline config.go]: It think it's better to add the leading `^` to the regexp and so it's exact match of `Host`.

Comment (jeloou) [reply config.go]: something like this?
```
^Host\\b\\s*=?\\s*(.+)
```

Comment (mrkschan) [reply config.go]: The one mentioned on https://github.com/crosbymichael/slex/pull/28#discussion_r155523999

Comment (jeloou) [reply config.go]: done, thanks!

**Comment by mrkschan** [comment]
Thanks @jeloou .

**Pull request merged**